src/orders/OrderProcessor.sol

```diff
@@ -794,7 +794,7 @@ contract OrderProcessor is
             IDShare(order.assetToken).mint(requester, unfilledAmount);
         } else {
             // Return unfilled
-            IERC20(order.paymentToken).safeTransferFrom(msg.sender, requester, unfilledAmount);
+            IERC20(order.paymentToken).safeTransfer(requester, unfilledAmount);
```

Comment:
Revert all changes to OrderProcessor. Only fees should be escrowed on this contract.